androidApp/src/main/java/com/mbta/tid/mbta_app/android/location/LocationDataManager.kt

```diff
@@ -0,0 +1,148 @@
+package com.mbta.tid.mbta_app.android.location
+
+import android.Manifest
+import android.annotation.SuppressLint
+import android.content.IntentSender
+import android.location.Location
+import android.os.Looper
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.DisposableEffect
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.platform.LocalContext
+import androidx.lifecycle.Lifecycle
+import androidx.lifecycle.LifecycleEventObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.compose.LocalLifecycleOwner
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.isGranted
+import com.google.accompanist.permissions.rememberMultiplePermissionsState
+import com.google.android.gms.common.api.ResolvableApiException
+import com.google.android.gms.location.LocationListener
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.location.LocationSettingsRequest
+import com.google.android.gms.location.Priority
+import com.mbta.tid.mbta_app.android.util.LocalActivity
+import com.mbta.tid.mbta_app.android.util.LocalLocationClient
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+open class LocationDataManager {
```

Comment:
This has to be `open` so that the `MockLocationDataManager` can inherit from it. I could've made `LocationDataManager` an interface instead, but I don't like the pattern of making everything a redundant interface; that said, we already do that for all our dependency injection repositories, so if you'd rather I also do that here for consistency I could make that change.

Reply:
Nah, I didn't realize it was for testing. That's fine IMO